### DISCUSSION TITLE
Are you used an "official" Turkish coding page?

### DISCUSSION THREAD

**Discussion by board707** [HakkanR/DMD2TUR#3] [General]
Merhaba :)

You mapped a specific national characters to the code range 0x14 -0x1F. Is this an official Turkish code table or you invented it just for your library?

I have a [DMD like library](https://github.com/board707/DMD_STM32) with Unicode character supports too. It designed mostly for Cyrillic fonts, but can be used with almost any letter-based (non-hieroglyphic) language.  Can I try to use your font with my library?

**Comment by HakkanR** [comment]
Merhaba :)

Dmd2 library just used 8 bit chars, in total 256 chars is enought for it and not all of them is used, when you send a unicode char, I mean ie.  a 16 bit char like "ş" char in Turkish, print function of arduino, sends it as two parts => 0xc6(197) after 0x9f(159) and dmd try to print them one by one.
I know this 0xc6 is not a valid char to print, but it's important me to understand if following char is 0x9f, I catch that user wants to print "ş" char. I passes 0x6c char as null  (not send it to be printed but saved it to catch following char is one of my Turkish chars) and forward it font file and char index which I desing a glypt for my special Turkish char, this is my logic and my table, not official :)

Now, your way to go can be like mine. Test your chars one by one, detect first byte of them and catch second ones, create your char in font and forward your char print requests to char index of your chars in font file.

I 've tried to be as clean as possible. I hope it can be understandable :)

**Comment by board707** [comment]
Thanks for your answer

> Now, your way to go can be like mine. Test your chars one by one

No, I use a different approach in my library. I'm creating a separate font file for national characters. It uses char codes that match the low byte of the UTF encoding. This allows you not to re-encode font files after converting it from TTF font.
My library works with monochrome(HUB12) and RGB (HUB75) matrices under the control of STM32 and RP2040 boards.
It will be interesting to add Turkish to my library - can I use your fonts?

**Reply by HakkanR** [reply]
Yes, it's an open source and you can use it freely.

**Comment by board707** [comment]
Thank you

**Comment by board707** [comment]
Example video : your font with my library
https://youtu.be/hsDohmxn968
Sorry for video quality
